utils/wasm-gen/src/generator/entry_points.rs

```diff
@@ -164,8 +166,31 @@ impl<'a, 'b> EntryPointsGenerator<'a, 'b> {
             (module, func_type)
         });
 
-        let export_body_instructions =
+        let mut export_body_instructions =
             self.generate_export_body(export_body_call_idx, export_body_call_func_type)?;
+
+        // after initializing the program, we will write about this in a special pointer
+        if name == "init" {
+            if let Some(memory_size_pages) = self.module.initial_mem_size() {
+                let mem_size = Into::<WasmPageCount>::into(memory_size_pages).memory_size();
+
+                let upper_limit = mem_size.saturating_sub(1) as i32;
+                let wait_called_ptr = upper_limit.saturating_sub(50);
+                let init_called_ptr = wait_called_ptr + mem::size_of::<bool>() as i32;
```

Comment:
These ptrs layouts are known in advance. I would move this code (and the same in _invocator.rs_) to some function/method.